### DISCUSSION TITLE
New fields for Technical Review of Proposals

### DISCUSSION THREAD

**Discussion by gnyiri** [UserOfficeProject/user-office-core#645] [Ideas]
Hi there,

at ELI-ERIC we would need some new attributes for Technical Reviews (e.g. Cost of development, Lab use, Beam time, ...) for the next Call (September 2024). 

What is the best approach to add these new fields if the other parties (ESS/STFC) are not interested or we cannot agree on the set of new fields in the short term? Even if we customize the backend through dep. injection, the frontend cannot be specialized (?).  

My ideas to proceed:

- update DB and backend to fit our needs but put the new fields in an expandable section on the UI (such as the Internal Reviews component)
- update DB and backend to fit our needs but add a new Feature 'Detailed Technical Review' to enable/disable the new fields on UI (User Office / Feature page)
- make Technical Reviews configurable through Templates (this seems to be a long term solution)

Any thoughts?

Thanks in advance,
Gergely

**Comment by yoganandaness** [comment]
@gnyiri Appreciate your time and your suggestion. I think it would be a good idea to go on dealing with Templates rather than new DB fields. This would be be a long term solution and also a flexible approach.

**Reply by gnyiri** [reply]
Thanks for your answer. Can we come up with a rough estimation (in man weeks) of implementing the Templates (incl. DB, backend, frontend + testing)? It would be really fine to know if its feasible for our next Call scheduled at September.

**Reply by bolmsten** [reply]
I would estimate roughly two weeks but this is with a developer that is experienced in the platform.  I know that STFC is interested in the same feature so the easiest is for you to sync with them.
